wgpu-types/src/lib.rs

```diff
@@ -3405,10 +3411,13 @@ impl TextureFormat {
         } else {
             basic
         };
-        let bgra8unorm = if device_features.contains(Features::BGRA8UNORM_STORAGE) {
-            attachment | TextureUsages::STORAGE_BINDING
+        let (bgra8unorm_f, bgra8unorm) = if device_features.contains(Features::BGRA8UNORM_STORAGE) {
+            (
+                msaa_resolve | TextureFormatFeatureFlags::STORAGE_WRITE_ONLY,
```

Comment:
With the changes in this PR, all the formats below that support `TextureUsages::STORAGE_BINDING` need at least one of the `TextureFormatFeatureFlags::STORAGE_*` flags to be enabled to be used as storage textures.

`wgpu::Features::TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES` bypasses `guaranteed_format_features` and that's why it gets the ray tracing example and one of the player tests passing.

With this PR it is effectively impossible to use/bind a storage texture (unless you enable `TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES`; which should not be required).

Running `cargo run --bin wgpu-examples storage_texture` results in:

> wgpu error: Validation Error
> 
> Caused by:
>   In Device::create_bind_group
>     The adapter does not support write access for storage textures of format Rgba8Unorm